src/tree/tenants/TenantTreeItem.ts

```diff
@@ -0,0 +1,43 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.md in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import { GenericItemOptions } from "../GenericItem";
+import { ResourceGroupsItem } from '../ResourceGroupsItem';
+
+export interface TenantItemOptions extends GenericItemOptions {
+    readonly checkboxState?: vscode.TreeItemCheckboxState;
+}
+
+export class TenantTreeItem implements ResourceGroupsItem {
+    constructor(public readonly label: string, public tenantId: string, public readonly account: vscode.AuthenticationSessionAccountInformation, private readonly options?: TenantItemOptions) {
+    }
+
```

Comment:
Nit: maybe this was a design choice, but you could pass the whole tenant object here instead of requesting the label and tenantId here. Then you can set tenantId and label based on the tenant object in the contructor.